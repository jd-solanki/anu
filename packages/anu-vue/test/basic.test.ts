import { mount } from '@vue/test-utils'
import { expect, test } from 'vitest'
import Hello from './Hello.vue'

test('mount component', async () => {
  expect(Hello).toBeTruthy()

  const wrapper = mount(Hello, {
    props: {
      count: 4,
    },
  })

  expect(wrapper.text()).toContain('4 x 2 = 8')
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('button').trigger('click')

  expect(wrapper.text()).toContain('4 x 3 = 12')

  await wrapper.get('button').trigger('click')

  expect(wrapper.text()).toContain('4 x 4 = 16')
})
