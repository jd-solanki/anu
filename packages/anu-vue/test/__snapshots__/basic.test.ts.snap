// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component 1`] = `
"<div>4 x 2 = 8</div>
<button> x1 </button>"
`;
